java-connectors/kafka-connect-azure-servicebus/src/main/java/io/lenses/streamreactor/connect/azure/servicebus/source/TaskToReceiverBridge.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.azure.servicebus.source;
+
+import io.lenses.kcql.Kcql;
+import io.lenses.streamreactor.connect.azure.servicebus.config.AzureServiceBusConfigConstants;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+
+/**
+ * Bridge between Receivers and Connector's Task for Azure Service Bus.
+ */
+@Slf4j
+public class TaskToReceiverBridge {
+
+  private static final int INITIAL_RECORDS_TO_COMMIT_SIZE = 500;
```

Comment:
Should this be configurable via KCQL?

Reply:
I don't think so, as it'll resize if needed (it's only INITIAL). 

If you think otherwise, then sure.

Reply:
In all honesty I'm not sure what it is - can we add some javadoc around it to explain in more detail?

Reply:
I mean: it's used as an initial size to initialize `recordsToCommit` in constructors:
`recordsToCommitMap = new ConcurrentHashMap<>(INITIAL_RECORDS_TO_COMMIT_SIZE);`

Maybe I can even give it a better name, but what would it be?

Reply:
It's ok, happy to leave it as it is.  Thanks for answering my question 🥇